source/main.cc

```diff
@@ -616,6 +616,20 @@ run_simulator(const std::string &raw_input_as_string,
 
           std::ofstream file(output_directory + "original.prm");
           file << raw_input_as_string;
+
+          // If using the Geodynamic World Builder, create output/original.wb
+          // containing the exact file used to create the world:
+          std::string world_builder_file = prm.get("World builder file");
+          if (world_builder_file != "")
+            {
+              // TODO: We just want to make a copy of the file, but to do this
+              // platform-independently we need the C++17 filesystem header.
+              // Update this when we require C++17
+              std::ifstream  wb_source(world_builder_file, std::ios::binary);
+              std::ofstream  wb_destination(output_directory + "original.wb",   std::ios::binary);
```

Comment:
why binary? these files are text files or not?

Reply:
I don't think it really matters. My rational was (besides that was what an exampled I find used) that since we need to read it into a stream and directly out into another stream, we don't need it to be converted to text mode by the stream and it can just copy the bites. But am not sure if this is how it really works internally and I don't really think it matters.

From https://en.cppreference.com/w/cpp/io/c/FILE#Binary_and_text_modes:
> A binary stream is an ordered sequence of characters that can transparently record internal data. Data read in from a binary stream always equal the data that were earlier written out to that stream, except that an implementation is allowed to append an indeterminate number of null characters to the end of the stream. A wide binary stream doesn't need to end in the initial shift state.

Reply:
This was @MFraters suggested edits so he can correct me if I'm wrong. To my understanding setting `std::ios::binary` doesn't define the file format, but that you want to copy the `world_builder_file` as it exists in memory to the output file without modifying the contents of `world_builder_file`.

This code does work as expected and outputs the contents of the input worldbuilder file exactly as it was input into ASPECT.

Reply:
Oh I didn't see that you had already replied Menno 😄

Reply:
I agree that there is no harm in doing it as binary (unmodified) data.